### ISSUE TITLE
Cargo build recompiles when using `rustc -- -Z no-trans`

### ISSUE THREAD

**Issue by antoyo** [rust-lang/cargo#3204]
Hello.

When I use `cargo rustc -- -Z -no-trans`, the following `cargo build` will recompile even thought I did not changed the source code.

Here is the commands that I ran:

```
$ cargo build
   Compiling test v0.1.0 (file:///path/to/project)
    Finished debug [unoptimized + debuginfo] target(s) in 0.40 secs
$ cargo rustc -- -Z no-trans
   Compiling test v0.1.0 (file:///path/to/project)
    Finished debug [unoptimized + debuginfo] target(s) in 0.17 secs
$ cargo build
   Compiling test v0.1.0 (file:///path/to/project)
    Finished debug [unoptimized + debuginfo] target(s) in 0.40 secs
```

The second `cargo build` should not recompile the project.

Thanks to fix this issue.

**Comment by alexcrichton** [comment]
Thanks for the report! Right now this is expected, however, due to how Cargo tracks `cargo rustc` and flags and such. Cargo has no knowledge of rustc flags, so it doesn't understand that `-Z no-trans` doesn't actually affect compilation, so the second `cargo build` to Cargo has a different set of flags being passed to the compiler, so the target is recompiled.

**Comment by xftroxgpx** [comment]
There's another side-effect to using `-Zno-trans`(which I'm using within `vim` when file gets saved).

Steps to repro:
1. change only the contents of `build.rs` file (or just `touch build.rs` ? not tested)
2. run `cargo rustc -- -Zno-trans`
(no errors, presumably)
3. run `cargo build`
(nothing gets built because of 2.!)

What happened? 
Step 2. has updated `build.rs` output (eg. this file `target/debug/build/recompile_self-feb93e46b6e91788/output`), 
so when I try step 3. it will NOT build anything.

Workaround is step 2.5.  `touch build.rs`

What's a better way?

**Comment by ehuss** [comment]
Closing as no longer applicable.
`-Z no-trans` was removed a while ago.
The issue @xftroxgpx mentions is fixed by #6720.
